main.rb

```diff
@@ -26,66 +33,54 @@ def h(text)
   end
 end
 
-# ------------------------- routing
-# top
 get '/' do
   redirect '/memos'
 end
 
 get '/memos' do
-  memos = files.map { |file| get_memo(file) }
-  @memos = memos.sort_by { |f| f['time'] }
+  @memos = memos.sort_by { |memo| memo['created_at'] }
   @title = 'トップ'
   erb :index
 end
 
-# new
 get '/memos/new' do
   @title = '入力画面'
   erb :new
 end
 
-# create
 post '/memos' do
-  memo = { 'id' => SecureRandom.hex, 'title' => params['title'], 'content' => params['content'], 'time' => Time.now }
-  File.open("./data/#{memo['id']}.json", 'w') do |f|
-    JSON.dump(memo, f)
-  end
+  memo = { 'id' => build_id, 'title' => params['title'], 'content' => params['content'], 'created_at' => Time.now }
+  create_memo(memo)
   redirect '/memos'
 end
 
-# show
-get '/memos/:id' do
-  File.exist?(memo) ? @memo = get_memo(memo) : (redirect to('not_found'))
+get '/memos/:id' do |id|
+  File.exist?(json_file(id)) ? @memo = get_memo(json_file(id)) : (redirect to('not_found'))
   @title = @memo['title']
-  @id = @memo['id']
-  @content = @memo['content']
   erb :show
 end
 
-# edit
-get '/memos/:id/edit' do
-  @memo = get_memo(memo)
+get '/memos/:id/edit' do |id|
+  @memo = get_memo(json_file(id))
```

Comment:
```suggestion
  @memo = get_memo(id)
```

となるように設計した方がコードがスッキリしそうです